frontend/src/features/rubricBuilder/CSVUpload.tsx

```diff
@@ -42,30 +58,201 @@ export const CSVUpload: React.FC<CSVUploadProps> = ({
               )
           );
         });
-        console.log("Validated CSV data:", results.data);
-        onDataChange(parsedData); // Pass validated data to parent
+
+        updateRubricWithCSVVersion1(parsedData);
+      },
+    });
+  };
+
+  const updateRubricWithCSVVersion1 = (data: CSVRow[]) => {
+    if (!rubric) return;
+
+    const newCriteria = data
+      .slice(1) // Skip header row
+      .map((row) => {
+        const criterionTitle = row[0]?.trim();
+        if (!criterionTitle) return null;
+
+        const criterion: Criteria = createCriterion(criterionTitle, "", 0, []);
+        for (let i = 1; i < row.length; i += 2) {
+          const points = Number(row[i]);
+          const description = row[i + 1]?.toString();
+          if (description) {
+            criterion.ratings.push(createRating(points, description));
+          }
+        }
+        criterion.updatePoints();
+        return criterion;
+      })
+      .filter(Boolean) as Criteria[];
+
+    setRubric((prevRubric) =>
+      prevRubric
+        ? {
+            ...prevRubric,
+            criteria: [...prevRubric.criteria, ...newCriteria],
+          }
+        : {
+            id: Date.now(), // Replace with a valid ID generator if needed
```

Comment:
I would just use `uuid()` like everything else for unique key generation

I guess this technically works, but seems like it could cause problems if two rubrics are made at the same time. Even if that's not an issue, rubric keys should at least be uniform across the app!